libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/offboard/MapboxOffboardRouter.kt

```diff
@@ -70,22 +69,48 @@ class MapboxOffboardRouter(
                     response: Response<DirectionsResponse>
                 ) {
                     directionRequests.remove(requestId)
+                    val url = call.request().url.toUrl()
                     val routes = response.body()?.routes()
+                    val code = response.code()
                     when {
-                        call.isCanceled -> callback.onCanceled()
-                        response.isSuccessful && !routes.isNullOrEmpty() -> {
-                            callback.onResponse(routes)
+                        call.isCanceled -> callback.onCanceled(routeOptions)
+                        response.isSuccessful -> {
+                            if (!routes.isNullOrEmpty()) {
+                                callback.onRoutesReady(routes)
+                            } else {
+                                callback.onFailure(
+                                    listOf(
+                                        RouterFailure(url, "routes list is empty", code)
+                                    ),
+                                    routeOptions
+                                )
+                            }
                         }
-                        else -> callback.onFailure(NavigationException(response.message()))
+                        else -> callback.onFailure(
+                            listOf(
+                                RouterFailure(url, response.errorBody()?.string().toString(), code)
```

Comment:
```suggestion
                                RouterFailure(url, response.errorBody()?.string()), code)
```
can be simplified?

Reply:
`ResponseBody#string` is nullable, that's why we need the extension